WordPress/WordPressTest/PushAuthenticationManagerTests.swift

```diff
@@ -0,0 +1,131 @@
+    }
+    
+    func expiredPushNotificationDictionary() -> NSDictionary {
+       return ["expires": NSTimeInterval(3)]
+    }
+    
+    func validPushAuthenticationDictionary() -> NSMutableDictionary {
+        return ["push_auth_token" : "token", "aps" : [ "alert" : "an alert"]]
+    }
+    
+    func testHandlePushAuthenticationNotificationShowsTheLoginExpiredAlertIfNotificationHasExpired(){
+        pushAuthenticationManager!.handlePushAuthenticationNotification(expiredPushNotificationDictionary())
+        
+        XCTAssertTrue(mockAlertViewProxy.showWithTitleCalled, "Should show the login expired alert if the notification has expired")
+        XCTAssertEqual(mockAlertViewProxy.titlePassedIn!, NSLocalizedString("Login Request Expired", comment:""), "")
+    }
+    
+    func testHandlePushAuthenticationNotificationDoesNotShowTheLoginExpiredAlertIfNotificationHasNotExpired(){
+        pushAuthenticationManager!.handlePushAuthenticationNotification([:])
+        
+        XCTAssertFalse(mockAlertViewProxy.showWithTitleCalled, "Should not show the login expired alert if the notification hasn't expired")
+    }
+    
+    func testHandlePushAuthenticationNotificationWithBlankTokenDoesNotShowLoginVerificationAlert(){
+        var pushNotificationDictionary = validPushAuthenticationDictionary()
+        pushNotificationDictionary.removeObjectForKey("push_auth_token")
+        
+        pushAuthenticationManager!.handlePushAuthenticationNotification(pushNotificationDictionary)
+        
+        XCTAssertFalse(mockAlertViewProxy.showWithTitleCalled, "Should not show the login verification")
+    }
+    
+    func testHandlePushAuthenticationNotificationWithBlankMessageDoesNotShowLoginVerificationAlert(){
+        pushAuthenticationManager!.handlePushAuthenticationNotification(["push_auth_token" : "token"])
+        
+        XCTAssertFalse(mockAlertViewProxy.showWithTitleCalled, "Should not show the login verification")
+    }
+    
+    func testHandlePushAuthenticationNotificationWithValidDataShouldShowLoginVerification() {
+        pushAuthenticationManager!.handlePushAuthenticationNotification(validPushAuthenticationDictionary())
+        
+        XCTAssertTrue(mockAlertViewProxy.showWithTitleCalled, "Should show the login verification")
+        XCTAssertEqual(mockAlertViewProxy.titlePassedIn!, NSLocalizedString("Verify Login", comment: ""), "")
+    }
+    
+    func testHandlePushAuthenticationNotificationShouldAttemptToAuthorizeTheLoginIfTheUserIndicatesTheyWantTo() {
+        pushAuthenticationManager!.handlePushAuthenticationNotification(validPushAuthenticationDictionary())
+        let alertView = UIAlertView()
+        mockAlertViewProxy.tapBlockPassedIn?(alertView, 1)
+        
+        XCTAssertTrue(mockPushAuthenticationService.authorizedLoginCalled, "Should have attempted to authorize the login")
+    }
+    
+    func testHandlePushAuthenticationNotificationWhenAttemptingToLoginShouldAttemptToRetryTheLoginIfItFailed() {
```

Comment:
Not really sure if that should go as a comment, since the `handlePushAuthenticationNotification` in fact shows an AlertView, and the retry itself is down the flow.

That was one of my biggest questions... testing private methods!. Feels a little bit like black box testing (and the only possible way around i could come up is: just switch it to Public, which is an awful idea).